libr/bin/format/elf/elf.c

```diff
@@ -83,17 +84,49 @@ static int init_ehdr(struct Elf_(r_bin_elf_obj_t) *bin) {
 #endif
 	bin->endian = (e_ident[EI_DATA] == ELFDATA2MSB)? 1: 0;
 	memset (&bin->ehdr, 0, sizeof (Elf_(Ehdr)));
-	len = r_buf_fread_at (bin->b, 0, (ut8*)&bin->ehdr,
-#if R_BIN_ELF64
-		bin->endian?"16c2SI3LI6S":"16c2si3li6s",
-#else
-		bin->endian?"16c2S5I6S":"16c2s5i6s",
-#endif
-			1);
+
+	len = r_buf_read_at (bin->b, 0, ehdr, sizeof (Elf_(Ehdr)));
 	if (len == -1) {
 		eprintf ("Warning: read (ehdr)\n");
 		return false;
 	}
+	memcpy (&bin->ehdr.e_ident, &ehdr[0], 16);
+	i = 16;
+	bin->ehdr.e_type = r_read_ble16 (&ehdr[i], bin->endian);
```

Comment:
It's likely I will try to remove r_buf_fread_at() completely, if @radare agrees.